### ISSUE TITLE
base image for binary R CRAN packages and RStudio server R4.0

### ISSUE THREAD

**Issue by Al-Murphy** [rocker-org/rocker-versioned2#161]
Hi,

I am trying to get a rocker base image for R 4.0 that can install binaries packages and has RStudio server available. I note that rocker-org/rocker-versioned2 is based on Ubuntu LTS so I think the binaries should be available, is this the case? I also noted:

>We use the RStudio Package Manager (RSPM) as our default CRAN mirror. RSPM serves compiled Linux binaries of R packages and greatly speeds up package installs.

What base image should I thus call to get these binaries? I have been using `FROM rocker/r-ubuntu` to get binaries but RStudio isn't available. I have tried `FROM rocker/rstudio` but binaries aren't available here.

My apologies if this isn't clear, I am relatively new to rocker!

**Comment by eddelbuettel** [comment]
With apologies about the state of affairs and the vast number of existing Dockerfiles,  your best bet for an overview is still our [R Journal paper](https://journal.r-project.org/archive/2017/RJ-2017-065/index.html) and its [pdf](https://journal.r-project.org/archive/2017/RJ-2017-065/RJ-2017-065.pdf) version. 

In essence, we have a 'base stack' (rocker/r-base, which also serves as r-base for all of Docker) plus various add-ons on top of that.  My two current favourites are `r-bspm` (using `bspm` to install R packages via `install.packages()` as `.deb` binaries) and `r-rspm` (same, using the wider set of RSPM which however do not have full depends).   And we have the 'versioned(2)' stack with a focus on reproducibility by freezing versions to dates.  Only that one has RStudio.  (And then these days we also have a ML stack closer to the versioned stack....)

So in short you could just derive an 'RStudio' base from one of the first one and add installation of the RStudio binary.  I do the latter locally with two scripts `getRStudioDesktop.r` and `getRStudioServer.` (from [here](https://github.com/eddelbuettel/littler/tree/master/inst/examples)) which I run somewhat frequently on Ubuntu (for which these scripts have the default setup because, well, I use them that way).  Just used them yesterday to get 1.5.42.  You could alter them to get the released rather than nightlies too.  

But as we sometimes state here, we are volunteers in this project so we sadly cannot cover all possibly permutations and variants. By covering the bases we hope to provide building blocks for tasks like the one you describe.  Hope this helps somewhat.

**Comment by cboettig** [comment]
Hi @Al-Murphy `rocker/rstudio` should absolutely be installing binaries from RStudio's RSPM service by default.  e.g. I just tested the following:

```
docker run --rm -ti rocker/rstudio install2.r purrr
also installing the dependencies ‘magrittr’, ‘rlang’

trying URL 'https://packagemanager.rstudio.com/all/__linux__/focal/latest/src/contrib/magrittr_2.0.1.tar.gz'
Content type 'binary/octet-stream' length 232440 bytes (226 KB)
==================================================
downloaded 226 KB

trying URL 'https://packagemanager.rstudio.com/all/__linux__/focal/latest/src/contrib/rlang_0.4.11.tar.gz'
Content type 'binary/octet-stream' length 1393952 bytes (1.3 MB)
==================================================
downloaded 1.3 MB

trying URL 'https://packagemanager.rstudio.com/all/__linux__/focal/latest/src/contrib/purrr_0.3.4.tar.gz'
Content type 'binary/octet-stream' length 426832 bytes (416 KB)
==================================================
downloaded 416 KB

* installing *binary* package ‘magrittr’ ...
* DONE (magrittr)
* installing *binary* package ‘rlang’ ...
* DONE (rlang)
* installing *binary* package ‘purrr’ ...
* DONE (purrr)

The downloaded source packages are in
        ‘/tmp/downloaded_packages’
```

Note the URLs point to RSPM and install binary packages.  IF that's not what you see, please let us know.  

As Dirk says, `rocker/rstudio` sits in the versioned stack, and is based on `r-ver`, which as you saw is based on `ubuntu:20.04`, see the [README](https://github.com/rocker-org/rocker-versioned2#readme).  Apologies the stacks are confusing, it's a lot of images with a lot of history :grimacing: but we could no doubt document things better!

**Comment by Al-Murphy** [comment]
Hey @eddelbuettel @cboettig, firstly thank you for your massive contribution by making something as useful as rocker, I did not mean to come across that I thought the docker images were unclear. I think it's just a lack of understanding on my own part!

I tried the `rocker/rstudio` approach but got the following error:
```
failed to solve with frontend dockerfile.v0: failed to create LLB definition: dockerfile parse error line 8: unknown instruction: INSTALL2.R
```

Here's the relevant part of the `Dockerfile`:
```
## ubuntu 20.04
FROM rocker/rstudio

install2.r purrr
```

**Comment by eddelbuettel** [comment]
A Dockerfile is _mostly shell-script alike_ but not identical.  Pick any Docker tutorial and experiment, and/or read our sources.  What you have there is lacking _e.g._ the `RUN` preceding the command and likely more.

**Comment by Al-Murphy** [comment]
My apologies silly error, that you both for your help on this, I really appreciate your time!
